### DISCUSSION TITLE
A big thank you for the migration guide

### DISCUSSION THREAD

**Discussion by sirhcel** [py-pdf/pypdf#1883] [Show and tell]
Yesterday, I had to upgrade PyPDF2 from 1.x to 3.x in a project. I had never touched PyPDF2 before and barely worked with the other project beforehand.

The [migration guide in the docs](https://pypdf2.readthedocs.io/en/latest/user/migration-1-to-2.html) and the code annotations brought me on track within minutes. This was one of the fastest and most pleasureful migrations I've done so far. I can't share the code but I can share this experience. ❤️ 

Thank you all very much for you love and effort!

**Comment by MartinThoma** [comment]
Thank you, this is much appreciated :hugs:
